src/components/group-common.js

```diff
@@ -104,76 +106,103 @@ export const NewUserModal = ajaxCaller(class NewUserModal extends Component {
   }
 
   render() {
-    const { adminLabel, userLabel, title, onDismiss, footer } = this.props
-    const { userEmail, roles, suggestions, submitting, submitError } = this.state
-
-    const errors = validate({ userEmail }, { userEmail: { email: true } })
+    const { adminLabel, userLabel, title, onDismiss, footer, inviteFunction } = this.props
+    const { confirmAddUser, userEmail, roles, suggestions, submitting, submitError, checkingUser } = this.state
+    const emailConstraints = { format: /\w+@(gmail\.com|broadinstitute\.org)$/ }
+    const errors = validate({ userEmail }, { userEmail: emailConstraints })
     const isAdmin = _.includes(adminLabel, roles)
 
-    const userEmailInvalid = !!validate({ userEmail }, { userEmail: { email: true } })
+    const userEmailInvalid = !!validate({ userEmail }, { userEmail: emailConstraints })
     const canAdd = value => value !== userEmail || !userEmailInvalid
 
-    return h(Modal, {
-      onDismiss,
-      title,
-      okButton: h(ButtonPrimary, {
-        tooltip: Utils.summarizeErrors(errors),
-        onClick: () => this.submit(),
-        disabled: errors
-      }, ['Add User'])
-    }, [
-      h(RequiredFormLabel, ['User email']),
-      h(AutocompleteSearch, {
-        autoFocus: true,
-        value: userEmail,
-        onChange: v => this.setState({ userEmail: v }),
-        renderSuggestion: suggestion => div({ style: styles.suggestionContainer }, [
-          div({ style: { flex: 1 } }, [
-            !canAdd(suggestion) && h(TooltipTrigger, {
-              content: 'Not a valid email address'
-            }, [
-              icon('warning-standard', { style: { color: colors.danger(), marginRight: '0.5rem' } })
-            ]),
-            suggestion
-          ])
-        ]),
-        onSuggestionSelected: selection => {
-          this.setState({ userEmail: selection })
-        },
-        onKeyDown: e => {
-          // 27 = Escape
-          if (e.which === 27 && !!userEmail) {
-            this.setState({ userEmail: '' })
-            e.stopPropagation()
+    return Utils.cond(
+      [confirmAddUser, h(Modal, {
+        title: 'This User Is Not Registered',
+        okButton: h(ButtonPrimary, {
+          onClick: async () => {
+            this.setState({ submitting: true })
+            await inviteFunction(userEmail)
```

Comment:
This modal is also used on the billing page, so without modifying it there as well, this would crash. That being said, does this really need to be passed in? Is there more than one way we add people?

Also, we should verify that it's desirable to have this whole enhanced flow on the billing page as well.

Reply:
I spoke with William, he asked to scope this to the groups page only. I added a prop, defaulting to false, that will bring up the additional confirmation dialog.

He did request that the client side email validation remain in both billing and groups for consistency